### PR TITLE
Expose probe_dns_duration_seconds metric

### PR DESCRIPTION
**Pull request by mem**
This follows the same pattern as probe_http_duration_seconds and
probe_icmp_duration_seconds: it captures the time it takes to perform
the actual request, excluding the time it takes to set up things.
probe_duration_seconds includes the time to perform the request as well
as all the time it takes to set it up. probe_dns_lookup_time_seconds
refers to the time it takes to resolve the target's address, so it
doesn't capture the time it takes to make one request for that target.

Signed-off-by: Marcelo E. Magallon <marcelo.magallon@grafana.com>

### CODE REVIEW

File: prober/dns.go [hunk 1]
```diff
@@ -246,7 +251,14 @@ func ProbeDNS(ctx context.Context, target string, module config.Module, registry
 	level.Info(logger).Log("msg", "Making DNS query", "target", targetIP, "dial_protocol", dialProtocol, "query", module.DNS.QueryName, "type", qt, "class", qc)
 	timeoutDeadline, _ := ctx.Deadline()
 	client.Timeout = time.Until(timeoutDeadline)
+	// we don't use the rtt value returned fom client.Exchange because that
+	// includes only the time to exchange messages with the server _after_
+	// the connection is created. We would in principle have three phases:
+	// resolve (probe_dns_lookup_time_seconds), connect (request - rtt),
```
Comment (brian-brazil) [inline prober/dns.go:257]: To keep things similar to the metric in the other probes, we should break this out into its phases.

Comment (mem) [reply prober/dns.go:257]: OK, will do

Comment (mem) [reply prober/dns.go]: I added the phase label and added the three time measurements, as indicated in the original comment.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**